src/IterableCodeExtractor.php

```diff
@@ -223,7 +227,7 @@ function ( $file, $key, $iterator ) use ( $include, $exclude, $extensions ) {
 						return static::containsMatchingChildren( $file, $include );
 					}
 
-					return ( ( $inclusion_score >= $exclusion_score ) && $file->isFile() && in_array( $file->getExtension(), $extensions, true ) );
+					return $inclusion_score >= $exclusion_score;
```

Comment:
This has now changed the logic, as we would now return a score for non-Files as well.
```suggestion
					return $inclusion_score >= $exclusion_score && $file->isFile();
```
@swissspidy Am I missing something here?

Reply:
Nope, you're right.

Could we perhaps change the condition above to:

```php
if ( ! $file->isFile() || ( $file->isFile() && ! in_array( $file->getExtension(), $extensions, true ) ) ) {
	return false;
}
```

instead?

Reply:
The second `$file->isFile()` is redundant. If the first one succeeds, it stops there, if it fails, then `isFile()` is guaranteed to be true.

Reply:
So:
```php
if ( ! $file->isFile() || ! in_array( $file->getExtension(), $extensions, true ) ) {
	return false;
}
```